tests/v2/validation/provisioning/k3s/etcd_node_delete_test.go

```diff
@@ -0,0 +1,162 @@
+	session               *session.Session
+	client                *rancher.Client
+	ns                    string
+	k3skubernetesVersions []string
+	cnis                  []string
+	providers             []string
+	nodesAndRoles         []machinepools.NodeRoles
+}
+
+func (e *EtcdNodeDeleteTestSuite) TearDownSuite() {
+	e.session.Cleanup()
+}
+
+func (e *EtcdNodeDeleteTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	e.session = testSession
+
+	e.ns = defaultNS
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	e.k3skubernetesVersions = clustersConfig.K3SKubernetesVersions
+
+	e.cnis = clustersConfig.CNIs
+	e.providers = clustersConfig.Providers
+	e.nodesAndRoles = clustersConfig.NodesAndRoles
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(e.T(), err)
+
+	e.client = client
+}
+
+func (e *EtcdNodeDeleteTestSuite) isEtcdNodeRemoved(query url.Values, clusterName string, etcdNodeToDelete v1.SteveAPIObject) (int, error) {
+	noOfEtcdNodesAfterDeletion := 0
+
+	err := kwait.Poll(5*time.Second, 15*time.Minute, func() (done bool, err error) {
+		machines, err := e.client.Steve.SteveType(machineSteveResourceType).List(query)
+		if err != nil {
+			return false, err
+		}
+
+		noOfEtcdNodesAfterDeletion = 0
+		for _, machine := range machines.Data {
+			if machine.Labels[etcdLabel] == "true" && machine.Labels[clusterLabel] == clusterName {
+				if machine.Name == etcdNodeToDelete.Name {
+					return false, nil
+				}
+				noOfEtcdNodesAfterDeletion++
+			}
+		}
+		logrus.Info("new etcd node : ")
+		for _, machine := range machines.Data {
+			if machine.Labels[etcdLabel] == "true" && machine.Labels[clusterLabel] == clusterName {
+				logrus.Info(machine.Name)
+			}
+		}
+		return true, nil
+	})
+	return noOfEtcdNodesAfterDeletion, err
+}
+
+func (e *EtcdNodeDeleteTestSuite) TestEtcdDeleteNode() {
+	require.GreaterOrEqual(e.T(), len(e.providers), 1)
+	require.GreaterOrEqual(e.T(), len(e.cnis), 1)
+	require.GreaterOrEqual(e.T(), len(e.k3skubernetesVersions), 1)
+
+	subSession := e.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := e.client.WithSession(subSession)
+	require.NoError(e.T(), err)
+
+	for _, provider := range e.providers {
+		for _, k8sVersion := range e.k3skubernetesVersions {
+			for _, cni := range e.cnis {
+				provider := CreateProvider(provider)
+				logrus.Info("provisioning ", k8sVersion, " cluster using ", cni, " CNI..")
+				clusterResp := TestProvisioningK3SCluster(e.T(), client, provider, e.nodesAndRoles, k8sVersion)
+
+				query, err := url.ParseQuery("")
+				require.NoError(e.T(), err)
+
+				query["fieldSelector"] = []string{"metadata.namespace=" + e.ns}
+
+				machines, err := e.client.Steve.SteveType(machineSteveResourceType).List(query)
+				require.NoError(e.T(), err)
+
+				noOfEtcdNodesBeforeDeletion := 0
+				etcdNodeToDelete := v1.SteveAPIObject{}
+
+				logrus.Info("etcd nodes before deletion : ")
```

Comment:
nit; I am wondering if this is even worth having? Looking at the Jenkins logging, seems to be a bit excessive, I think having the etcd node deleting log is nice, but not sure if we want to have all the etcd nodes shown?

Reply:
agreed. Right after, the node to be deleted is shown, which looks good in the Jenkins log IMO. Listing all the nodes seemed a bit much to me

Reply:
Removed the logging for listing all nodes.